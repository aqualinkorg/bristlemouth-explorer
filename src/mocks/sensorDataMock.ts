import { SensorData } from 'src/helpers/types';

export const sensorDataMock: SensorData[] = [
  {
    latitude: 20.78995,
    longitude: -156.5069833,
    timestamp: '2023-09-13T20:04:45.000Z',
    sensorPosition: 1,
    units: '°C',
    value: 26.22,
    unit_type: 'temperature',
    data_type_name: 'sofar_temperature_12bits',
  },
  {
    latitude: 20.78995,
    longitude: -156.5069833,
    timestamp: '2023-09-13T20:04:45.000Z',
    sensorPosition: 2,
    units: '°C',
    value: 26.1,
    unit_type: 'temperature',
    data_type_name: 'sofar_temperature_12bits',
  },
  {
    latitude: 20.78995,
    longitude: -156.5069833,
    timestamp: '2023-09-13T20:24:45.000Z',
    sensorPosition: 1,
    units: '°C',
    value: 26.240000000000002,
    unit_type: 'temperature',
    data_type_name: 'sofar_temperature_12bits',
  },
  {
    latitude: 20.78995,
    longitude: -156.5069833,
    timestamp: '2023-09-13T20:24:45.000Z',
    sensorPosition: 2,
    units: '°C',
    value: 26.119999999999997,
    unit_type: 'temperature',
    data_type_name: 'sofar_temperature_12bits',
  },
  {
    latitude: 20.78995,
    longitude: -156.5069833,
    timestamp: '2023-09-13T20:44:45.000Z',
    sensorPosition: 1,
    units: '°C',
    value: 26.259999999999998,
    unit_type: 'temperature',
    data_type_name: 'sofar_temperature_12bits',
  },
  {
    latitude: 20.78995,
    longitude: -156.5069833,
    timestamp: '2023-09-13T20:44:45.000Z',
    sensorPosition: 2,
    units: '°C',
    value: 26.14,
    unit_type: 'temperature',
    data_type_name: 'sofar_temperature_12bits',
  },
  {
    latitude: 20.78995,
    longitude: -156.5069833,
    timestamp: '2023-09-13T21:04:45.000Z',
    sensorPosition: 1,
    units: '°C',
    value: 26.300000000000004,
    unit_type: 'temperature',
    data_type_name: 'sofar_temperature_12bits',
  },
  {
    latitude: 20.78995,
    longitude: -156.5069833,
    timestamp: '2023-09-13T21:04:45.000Z',
    sensorPosition: 2,
    units: '°C',
    value: 26.160000000000004,
    unit_type: 'temperature',
    data_type_name: 'sofar_temperature_12bits',
  },
  {
    latitude: 20.78995,
    longitude: -156.5069833,
    timestamp: '2023-09-13T21:24:45.000Z',
    sensorPosition: 1,
    units: '°C',
    value: 26.340000000000003,
    unit_type: 'temperature',
    data_type_name: 'sofar_temperature_12bits',
  },
  {
    latitude: 20.78995,
    longitude: -156.5069833,
    timestamp: '2023-09-13T21:24:45.000Z',
    sensorPosition: 2,
    units: '°C',
    value: 26.200000000000003,
    unit_type: 'temperature',
    data_type_name: 'sofar_temperature_12bits',
  },
];
