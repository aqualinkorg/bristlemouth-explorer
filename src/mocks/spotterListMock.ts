import { Spotter } from 'src/helpers/types';

export const spotterListMock: Spotter[] = [
  {
    name: 'Totoro',
    spotterId: 'SPOT-300434063450120',
  },
  {
    name: 'USA W01',
    spotterId: 'SPOT-0514',
  },
  {
    name: null,
    spotterId: 'SPOT-1034',
  },
];
