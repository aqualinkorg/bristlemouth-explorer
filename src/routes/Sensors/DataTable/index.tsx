import React from 'react';
import {
  Box,
  Collapse,
  IconButton,
  Paper,
  Stack,
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TableRow,
  Typography,
  styled,
} from '@mui/material';
import ExpandCircleDownIcon from '@mui/icons-material/ExpandCircleDown';
import { useSelector } from 'react-redux';
import { sensorDataSelector } from 'src/store/spotters/spottersSlice';
import { SensorData } from 'src/helpers/types';

interface TableData {
  timestamp: string;
  sensorPosition: string;
  rawData: string;
}

interface RowProps {
  data: TableData;
}

function Row({ data }: RowProps) {
  const [open, setOpen] = React.useState(false);

  return (
    <>
      <TableRow sx={{ '& > *': { borderBottom: 'unset' } }}>
        <TableCell>
          <IconButton
            aria-label="expand row"
            size="small"
            onClick={() => setOpen(!open)}
          >
            {open ? (
              <ExpandCircleDownIcon
                color="primary"
                fontSize="small"
                style={{ transform: 'rotate(180deg)' }}
              />
            ) : (
              <ExpandCircleDownIcon color="primary" fontSize="small" />
            )}
          </IconButton>
        </TableCell>
        <TableCell component="th" scope="row">
          {data.timestamp}
        </TableCell>
        <TableCell align="right">{data.sensorPosition}</TableCell>
        <TableCell align="right">{data.rawData}</TableCell>
      </TableRow>
      <TableRow>
        <TableCell style={{ paddingBottom: 0, paddingTop: 0 }} colSpan={6}>
          <Collapse in={open} timeout="auto" unmountOnExit>
            <Box sx={{ margin: 1 }}>
              <Typography variant="h6" gutterBottom component="div">
                Extra Information
              </Typography>
            </Box>
          </Collapse>
        </TableCell>
      </TableRow>
    </>
  );
}

const PaperContainer = styled(Paper)(({ theme }) => ({
  width: '100vw',
  height: '60vh',
  margin: '1rem 0.5rem 0.5rem 0.5rem',
  padding: '1rem',
  borderRadius: theme.spacing(1),
}));

const StyledTableHead = styled(TableHead)(({ theme }) => ({
  backgroundColor: theme.palette.grey[100],
  borderRadius: theme.spacing(1),
}));

const HeaderTableCell = styled(TableCell)(() => ({
  fontWeight: 'bold',
}));

function transform(data: SensorData[]): TableData[] {
  return data.map((x) => ({
    timestamp: x.timestamp,
    sensorPosition: String(x.sensorPosition),
    rawData: x.value as string,
  }));
}

function DataTable() {
  const sensorData = useSelector(sensorDataSelector);

  return (
    <PaperContainer>
      <Stack gap="2rem" height="100%">
        <Typography variant="h6" fontWeight="bold">
          Data
        </Typography>

        <TableContainer>
          <Table size="small" stickyHeader>
            <StyledTableHead>
              <TableRow>
                <HeaderTableCell />
                <HeaderTableCell>Timestamp</HeaderTableCell>
                <HeaderTableCell align="right">Sensor Position</HeaderTableCell>
                <HeaderTableCell align="right">Encoded value</HeaderTableCell>
              </TableRow>
            </StyledTableHead>
            <TableBody>
              {transform(sensorData).map((data) => (
                <Row
                  key={`${data.timestamp}_${data.sensorPosition}`}
                  data={data}
                />
              ))}
            </TableBody>
          </Table>
        </TableContainer>
      </Stack>
    </PaperContainer>
  );
}

export default DataTable;
