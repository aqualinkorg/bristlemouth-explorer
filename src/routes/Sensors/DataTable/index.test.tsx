import { render } from '@testing-library/react';
import DataTable from '.';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { sensorDataMock } from 'src/mocks/sensorDataMock';

const mockStore = configureStore([]);

const store = mockStore({
  spotters: {
    list: [],
    sensorData: sensorDataMock,
    sensorDataLoading: false,
    spottersRequestLoading: false,
    error: null,
  },
});

store.dispatch = jest.fn();

it('renders as expected', () => {
  const { container } = render(
    <Provider store={store}>
      <DataTable />
    </Provider>,
  );
  expect(container).toMatchSnapshot();
});
