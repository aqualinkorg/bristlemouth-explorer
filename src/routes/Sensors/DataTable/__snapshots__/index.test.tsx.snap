// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-m1dl3w-MuiPaper-root"
  >
    <div
      class="MuiStack-root css-1p9m833-MuiStack-root"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-tjvwag-MuiTypography-root"
      >
        Data
      </h6>
      <div
        class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
      >
        <table
          class="MuiTable-root MuiTable-stickyHeader css-xn82ks-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-s6vh1v-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeSmall css-1hrvygi-MuiTableCell-root"
                scope="col"
              />
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeSmall css-1hrvygi-MuiTableCell-root"
                scope="col"
              >
                Timestamp
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignRight MuiTableCell-sizeSmall css-e30gri-MuiTableCell-root"
                scope="col"
              >
                Sensor position
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignRight MuiTableCell-sizeSmall css-e30gri-MuiTableCell-root"
                scope="col"
              >
                Encoded value
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root css-2blr9s-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
              >
                <button
                  aria-label="expand row"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeSmall css-1bptiuh-MuiSvgIcon-root"
                    data-testid="ExpandCircleDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 13.5L7.5 11l1.42-1.41L12 12.67l3.08-3.08L16.5 11 12 15.5z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </td>
              <th
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                scope="row"
              >
                2023-09-13T20:04:45.000Z
              </th>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-dv0mq-MuiTableCell-root"
              >
                1
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-dv0mq-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-fx27db-MuiTypography-root"
                >
                  26.22
                </p>
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                colspan="6"
                style="padding-bottom: 0px; padding-top: 0px;"
              />
            </tr>
            <tr
              class="MuiTableRow-root css-2blr9s-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
              >
                <button
                  aria-label="expand row"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeSmall css-1bptiuh-MuiSvgIcon-root"
                    data-testid="ExpandCircleDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 13.5L7.5 11l1.42-1.41L12 12.67l3.08-3.08L16.5 11 12 15.5z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </td>
              <th
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                scope="row"
              >
                2023-09-13T20:04:45.000Z
              </th>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-dv0mq-MuiTableCell-root"
              >
                2
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-dv0mq-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-fx27db-MuiTypography-root"
                >
                  26.1
                </p>
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                colspan="6"
                style="padding-bottom: 0px; padding-top: 0px;"
              />
            </tr>
            <tr
              class="MuiTableRow-root css-2blr9s-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
              >
                <button
                  aria-label="expand row"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeSmall css-1bptiuh-MuiSvgIcon-root"
                    data-testid="ExpandCircleDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 13.5L7.5 11l1.42-1.41L12 12.67l3.08-3.08L16.5 11 12 15.5z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </td>
              <th
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                scope="row"
              >
                2023-09-13T20:24:45.000Z
              </th>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-dv0mq-MuiTableCell-root"
              >
                1
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-dv0mq-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-fx27db-MuiTypography-root"
                >
                  26.240000000000002
                </p>
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                colspan="6"
                style="padding-bottom: 0px; padding-top: 0px;"
              />
            </tr>
            <tr
              class="MuiTableRow-root css-2blr9s-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
              >
                <button
                  aria-label="expand row"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeSmall css-1bptiuh-MuiSvgIcon-root"
                    data-testid="ExpandCircleDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 13.5L7.5 11l1.42-1.41L12 12.67l3.08-3.08L16.5 11 12 15.5z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </td>
              <th
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                scope="row"
              >
                2023-09-13T20:24:45.000Z
              </th>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-dv0mq-MuiTableCell-root"
              >
                2
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-dv0mq-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-fx27db-MuiTypography-root"
                >
                  26.119999999999997
                </p>
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                colspan="6"
                style="padding-bottom: 0px; padding-top: 0px;"
              />
            </tr>
            <tr
              class="MuiTableRow-root css-2blr9s-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
              >
                <button
                  aria-label="expand row"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeSmall css-1bptiuh-MuiSvgIcon-root"
                    data-testid="ExpandCircleDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 13.5L7.5 11l1.42-1.41L12 12.67l3.08-3.08L16.5 11 12 15.5z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </td>
              <th
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                scope="row"
              >
                2023-09-13T20:44:45.000Z
              </th>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-dv0mq-MuiTableCell-root"
              >
                1
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-dv0mq-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-fx27db-MuiTypography-root"
                >
                  26.259999999999998
                </p>
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                colspan="6"
                style="padding-bottom: 0px; padding-top: 0px;"
              />
            </tr>
            <tr
              class="MuiTableRow-root css-2blr9s-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
              >
                <button
                  aria-label="expand row"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeSmall css-1bptiuh-MuiSvgIcon-root"
                    data-testid="ExpandCircleDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 13.5L7.5 11l1.42-1.41L12 12.67l3.08-3.08L16.5 11 12 15.5z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </td>
              <th
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                scope="row"
              >
                2023-09-13T20:44:45.000Z
              </th>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-dv0mq-MuiTableCell-root"
              >
                2
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-dv0mq-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-fx27db-MuiTypography-root"
                >
                  26.14
                </p>
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                colspan="6"
                style="padding-bottom: 0px; padding-top: 0px;"
              />
            </tr>
            <tr
              class="MuiTableRow-root css-2blr9s-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
              >
                <button
                  aria-label="expand row"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeSmall css-1bptiuh-MuiSvgIcon-root"
                    data-testid="ExpandCircleDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 13.5L7.5 11l1.42-1.41L12 12.67l3.08-3.08L16.5 11 12 15.5z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </td>
              <th
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                scope="row"
              >
                2023-09-13T21:04:45.000Z
              </th>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-dv0mq-MuiTableCell-root"
              >
                1
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-dv0mq-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-fx27db-MuiTypography-root"
                >
                  26.300000000000004
                </p>
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                colspan="6"
                style="padding-bottom: 0px; padding-top: 0px;"
              />
            </tr>
            <tr
              class="MuiTableRow-root css-2blr9s-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
              >
                <button
                  aria-label="expand row"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeSmall css-1bptiuh-MuiSvgIcon-root"
                    data-testid="ExpandCircleDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 13.5L7.5 11l1.42-1.41L12 12.67l3.08-3.08L16.5 11 12 15.5z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </td>
              <th
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                scope="row"
              >
                2023-09-13T21:04:45.000Z
              </th>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-dv0mq-MuiTableCell-root"
              >
                2
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-dv0mq-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-fx27db-MuiTypography-root"
                >
                  26.160000000000004
                </p>
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                colspan="6"
                style="padding-bottom: 0px; padding-top: 0px;"
              />
            </tr>
            <tr
              class="MuiTableRow-root css-2blr9s-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
              >
                <button
                  aria-label="expand row"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeSmall css-1bptiuh-MuiSvgIcon-root"
                    data-testid="ExpandCircleDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 13.5L7.5 11l1.42-1.41L12 12.67l3.08-3.08L16.5 11 12 15.5z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </td>
              <th
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                scope="row"
              >
                2023-09-13T21:24:45.000Z
              </th>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-dv0mq-MuiTableCell-root"
              >
                1
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-dv0mq-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-fx27db-MuiTypography-root"
                >
                  26.340000000000003
                </p>
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                colspan="6"
                style="padding-bottom: 0px; padding-top: 0px;"
              />
            </tr>
            <tr
              class="MuiTableRow-root css-2blr9s-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
              >
                <button
                  aria-label="expand row"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeSmall css-1bptiuh-MuiSvgIcon-root"
                    data-testid="ExpandCircleDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 13.5L7.5 11l1.42-1.41L12 12.67l3.08-3.08L16.5 11 12 15.5z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </td>
              <th
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                scope="row"
              >
                2023-09-13T21:24:45.000Z
              </th>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-dv0mq-MuiTableCell-root"
              >
                2
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-dv0mq-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-fx27db-MuiTypography-root"
                >
                  26.200000000000003
                </p>
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                colspan="6"
                style="padding-bottom: 0px; padding-top: 0px;"
              />
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
